File: app/src/main/java/com/mudrichenko/evgeniy/flickrtestproject/ui/profile/ProfileFragment.java
package com.mudrichenko.evgeniy.flickrtestproject.ui.profile;

import android.os.Bundle;
import android.support.v4.view.PagerTitleStrip;
import android.support.v4.view.ViewPager;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.arellomobile.mvp.presenter.InjectPresenter;
import com.mudrichenko.evgeniy.flickrtestproject.R;
import com.mudrichenko.evgeniy.flickrtestproject.ui.BaseFragment;

public class ProfileFragment extends BaseFragment implements ProfileView, ViewPager.OnPageChangeListener {

    @InjectPresenter
    ProfilePresenter mProfilePresenter;

    ViewPager mViewPager;

    PagerTitleStrip mPagerTitleStrip;

    private final static String MAIN_MENU_FRAGMENT_SAVED_FRAGMENT = "main_menu_fragment_saved_fragment";

    private final static int DEFAULT_FRAGMENT_ID = 0;

    private final static int NUM_OF_PAGES = 4;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_profile, container, false);
        mViewPager = view.findViewById(R.id.view_pager);
        mPagerTitleStrip = view.findViewById(R.id.pager_title_strip);
        mViewPager.setAdapter(mProfilePresenter.getPagerAdapter(getChildFragmentManager()));
        mViewPager.setOffscreenPageLimit(NUM_OF_PAGES - 1);
        mViewPager.addOnPageChangeListener(this);
        if (savedInstanceState != null) {
            mProfilePresenter.setViewPagerCurrentPosition(savedInstanceState.getInt(MAIN_MENU_FRAGMENT_SAVED_FRAGMENT, DEFAULT_FRAGMENT_ID));
        }
        return view;
    }

    @Override
    public void onResume() {
        super.onResume();
        mViewPager.setCurrentItem(mProfilePresenter.getViewPagerCurrentPosition());
    }

    public static ProfileFragment newInstance() {
        ProfileFragment fragment = new ProfileFragment();
        return fragment;
    }

    @Override
    public void onSaveInstanceState(Bundle bundle) {
        super.onSaveInstanceState(bundle);
        bundle.putInt(MAIN_MENU_FRAGMENT_SAVED_FRAGMENT, mViewPager.getCurrentItem());
    }

    @Override
    public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {
        mProfilePresenter.onViewPagerScrolled(position, positionOffset);
    }

    @Override
    public void onPageSelected(int position) {

    }

    @Override
    public void onPageScrollStateChanged(int state) {

    }
}

File: app/src/main/java/com/mudrichenko/evgeniy/flickrtestproject/ui/profile/ProfileFragment.kt
package com.mudrichenko.evgeniy.flickrtestproject.ui.profile

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import androidx.viewpager.widget.PagerTitleStrip
import androidx.viewpager.widget.ViewPager

import com.arellomobile.mvp.presenter.InjectPresenter
import com.mudrichenko.evgeniy.flickrtestproject.R
import com.mudrichenko.evgeniy.flickrtestproject.ui.BaseFragment

class ProfileFragment : BaseFragment(), ProfileView, ViewPager.OnPageChangeListener {

    @InjectPresenter
    lateinit var mProfilePresenter: ProfilePresenter

    private lateinit var mViewPager: ViewPager

    private lateinit var mPagerTitleStrip: PagerTitleStrip

    private val MAIN_MENU_FRAGMENT_SAVED_FRAGMENT = "main_menu_fragment_saved_fragment"

    private val DEFAULT_FRAGMENT_ID = 0

    private val NUM_OF_PAGES = 4

    companion object {
        fun newInstance(): ProfileFragment {
            return ProfileFragment()
        }
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.fragment_profile, container, false)
        mViewPager = view.findViewById(R.id.view_pager)
        mPagerTitleStrip = view.findViewById(R.id.pager_title_strip)
        mViewPager.adapter = mProfilePresenter.getPagerAdapter(childFragmentManager)
        mViewPager.offscreenPageLimit = NUM_OF_PAGES - 1
        mViewPager.addOnPageChangeListener(this)
        if (savedInstanceState != null) {
            mProfilePresenter.viewPagerCurrentPosition = savedInstanceState.getInt(MAIN_MENU_FRAGMENT_SAVED_FRAGMENT, DEFAULT_FRAGMENT_ID)
        }
        return view
    }

    override fun onResume() {
        super.onResume()
        mViewPager.currentItem = mProfilePresenter.viewPagerCurrentPosition
    }

    override fun onSaveInstanceState(bundle: Bundle) {
        super.onSaveInstanceState(bundle)
        bundle.putInt(MAIN_MENU_FRAGMENT_SAVED_FRAGMENT, mViewPager.currentItem)
    }

    override fun onPageScrolled(position: Int, positionOffset: Float, positionOffsetPixels: Int) {
        mProfilePresenter.onViewPagerScrolled(position, positionOffset)
    }

    override fun onPageSelected(position: Int) {

    }

    override fun onPageScrollStateChanged(state: Int) {

    }

}
